src/libfetchers/tarball.cc

```diff
@@ -101,7 +100,7 @@ DownloadFileResult downloadFile(
             inAttrs,
             infoAttrs,
             *storePath,
-            locked);
+            false);
```

Comment:
Could we also just remove this? Or does something else call it with `true`?

Reply:
It's still used by the Mercurial fetcher, but it's removed in this followup PR: https://github.com/NixOS/nix/pull/10465/files